modules/ROOT/pages/indexes-for-nearest-neighbor.adoc

```diff
@@ -0,0 +1,431 @@
+| 0.9267876148223877
+
+| "Learning to Route in Similarity Graphs"
+| 0.9263730049133301
+
+2+d| Rows: 10
+
+|===
+
+The results are quite expected, with papers discussing graph-based nearest-neighbor searches.
+
+The most similar to this result is the query vector itself, which is to be expected as you queried the index with an indexed property.
+If the query vector itself is not wanted, you can use `WHERE score < 1` to remove equivalent vectors to the query vector.
+
+====
+
+[[indexes-vector-drop]]
+== Drop vector indexes
+
+A vector index is dropped by using the xref:indexes-for-search-performance.adoc#indexes-drop-an-index[same command as for other indexes], `DROP INDEX`.
+
+.+DROP INDEX+
+======
+
+In the following example, you drop the `abstract-embeddings` that you created previously:
+
+.Query
+[source,cypher]
+----
+DROP INDEX `abstract-embeddings`
+----
+
+.Result
+[queryresult]
+----
+Removed 1 index.
+----
+
+======
+
+[[indexes-vector-set]]
+== Set vector property on a node
+
+You can set a vector property, `LIST<FLOAT>`, on a node using the xref:clauses/set.adoc#set-set-a-property[`SET`] command:
+
+.Syntax for setting a vector property
+[source,cypher]
+----
+SET node.propertyKey = vector
+----
+
+However, Cypher stores the provided `LIST<FLOAT>` as a primitive array of {ieee-754} _double_ precision values.
+Valid vectors for use in the index must have components finitely representable in {ieee-754} _single_ precision.
+As a result, it takes up approximately twice as much space.
+
+The {link-procedures-reference}#procedure_db_create_setVectorProperty[`db.create.setVectorProperty`] procedure is provided to validate and set the property as an array of {ieee-754} single precision values.
+As a result, the store's space is approximately halved.
+
+.Signature for `db.create.setVectorProperty` to set a vector property
+[source, syntax, role="noheader", indent=0]
+----
+db.create.setVectorProperty(node :: NODE?, key :: STRING?, vector :: LIST? OF FLOAT?) :: (node :: NODE?)
+----
+
+Thus, the equivalent syntax for setting as the aforementioned:
+
+.Syntax for setting a vector _via_ a procedure
+[source,cypher]
+----
+CALL db.create.setVectorProperty(node, 'propertyKey', vector)
+----
+
+This is ideal for creating and setting new vector properties in the graph.
+Though existing properties can be re-set using this procedure to minimize store size, there is a cost in transaction log size until they are rotated away, especially in the case of record formats.
+
+[[indexes-vector-similarity]]
+== Supported similarity functions
+
+The choice of similarity function can vastly differ in which vectors are considered similar, and which are valid.
+The semantic meaning of the vector may itself dictate which similarity function to choose.
+Consult the documentation for the particular model used to generate vector embeddings.
+The documentation may suggest a preference for certain similarity functions.
```

Comment:
```suggestion
Consult the documentation for the particular model used to generate vector embeddings, as this may suggest a preference for certain similarity functions.
```

Reply:
Hey @parnmatt , which documentation do you refer to? Could you provide us with a link?